packages/agreement-process/src/services/agreementService.ts

```diff
@@ -367,9 +368,7 @@ export function agreementServiceBuilder(
         readModelService
       );
 
-      if (agreement.data.state === agreementState.draft) {
```

Comment:
removed redundant check on agreement state, because it was done by `assertSubmittableState` previously.